src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -448,16 +490,54 @@ static void create_stub_global_symbols(
         irep_idt class_id = instruction.args[0].get_string(ID_class);
         INVARIANT(
           !class_id.empty(), "get/putstatic should specify a class");
-        irep_idt identifier = id2string(class_id) + "." + id2string(component);
 
-        if(!symbol_table.has_symbol(identifier))
+        resolve_inherited_componentt::inherited_componentt referred_component =
+          get_inherited_component(
+            class_id,
+            component,
+            "java::" + id2string(parse_tree.parsed_class.name),
+            symbol_table,
+            class_hierarchy,
+            true);
+        if(!referred_component.is_valid())
         {
+          // Create a new stub global on an arbitrary incomplete ancestor of the
+          // class that was referred to. This is just a guess, but we have no
+          // better information to go on.
+          irep_idt add_to_class_id =
+            get_any_incomplete_ancestor(
+              class_id, symbol_table, class_hierarchy);
+
+          // If there are no incomplete ancestors to ascribe the missing field
+          // to, we must have an incomplete model of a class or simply a
+          // version mismatch of some kind. Normally this would be an error, but
+          // our models library currently triggers this error in some cases
+          // (notably java.lang.System, which is missing System.in/out/err).
+          // Therefore for this case we ascribe the missing field to the class
+          // it was directly referenced from, and fall back to initialising the
+          // field in __CPROVER_initialize, rather than try to create or augment
+          // a clinit method for a non-stub class.
+
+          bool no_incomplete_ancestors = add_to_class_id.empty();
+          if(no_incomplete_ancestors)
+          {
+            add_to_class_id = class_id;
+
+            log.warning() << "Stub static field " << component << " found for "
+                          << "non-stub type " << class_id << ". In future this "
+                          << "will be a fatal error." << messaget::eom;
```

Comment:
is there an issue for this already? if not, would it make sense to add an easier to find TODO here?

Reply:
PR is here: https://github.com/diffblue/models-library/pull/361
Once it's merged (and updated in test-gen) you should be able to replace the warning with an error. I added tests in my PR that showed the warning before the change in the model, and do not show it any more after the change.